pkg/agent/clientset.go

```diff
@@ -211,20 +217,32 @@ func (cs *ClientSet) sync() {
 	}
 }
 
-func (cs *ClientSet) connectOnce() error {
-	if !cs.syncForever && cs.serverCount != 0 && cs.ClientsCount() >= cs.serverCount {
+func (cs *ClientSet) ServerCount(ctx context.Context) int {
+	var serverCount int
+	if cs.serverCounter != nil {
+		serverCount = cs.serverCounter.Count(ctx)
+		if serverCount == 0 {
+			klog.Warningf("server lease counter could not find any leases")
+		}
+	} else {
+		serverCount = cs.lastReceivedServerCount
+	}
+
+	metrics.Metrics.SetServerCount(serverCount)
+	return serverCount
+}
+
+func (cs *ClientSet) connectOnce(ctx context.Context) error {
+	serverCount := cs.ServerCount(ctx)
+
+	if !cs.syncForever && serverCount != 0 && cs.ClientsCount() >= serverCount {
 		return nil
 	}
-	c, serverCount, err := cs.newAgentClient()
+	c, receivedServerCount, err := cs.newAgentClient()
 	if err != nil {
 		return err
 	}
-	if cs.serverCount != 0 && cs.serverCount != serverCount {
-		klog.V(2).InfoS("Server count change suggestion by server",
-			"current", cs.serverCount, "serverID", c.serverID, "actual", serverCount)
-
-	}
-	cs.serverCount = serverCount
+	cs.lastReceivedServerCount = receivedServerCount
```

Comment:
We should not be losing the server count changed warning.

Reply:
Added it back in!

Reply:
I can't seem to find it.

Reply:
https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/643/files#diff-f79c206ef287efc120666a024e5a23b09183ad983123a5cbd3fde793b261d0f8R228-R231